modules/openideal_user/src/EventSubscriber/OpenidealUserEventSubscriber.php

```diff
@@ -52,9 +102,35 @@ public static function getSubscribedEvents() {
     return [
       CKEditorMentionEvent::MENTION_FIRST => 'usersAreMentioned',
       ContentModerationEvents::STATE_CHANGED => 'stateChanged',
+      KernelEvents::RESPONSE => 'response',
     ];
   }
 
+  /**
+   * This method is called when the RESPONSE event is dispatched.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\FilterResponseEvent $event
+   *   Dispatched event.
+   */
+  public function response(FilterResponseEvent $event) {
+    if ($event->getRequest()->get('_route') == 'view.frontpage.front_page') {
+      // Set a cookie for anonymous user when
+      // visits front page for the first time.
+      if ($this->currentUser->isAnonymous() && !$event->getRequest()->cookies->has('OpenideaL_visitor_tour_front_page')) {
+        $cookie = new Cookie('OpenideaL.visitor.tour_front_page', '1', $this->time->getRequestTime() + 31536000, '/');
+        $response = $event->getResponse();
+        $response->headers->setCookie($cookie);
```

Comment:
Let's try to attach a cookie on FE level.
**Tour** module provides some events (drupalTourStarted, drupalTourStopped), so we can react accordingly.

In the subscriber, we need only to perform a redirect to the same page with GET parameter appended.

Reply:
Done